extensions-core/gce-extensions/src/main/java/org/apache/druid/indexing/overlord/autoscaling/gce/GCEAutoScaler.java

```diff
@@ -105,37 +96,37 @@ public static Compute createComputeService() throws IOException, GeneralSecurity
     HttpRequestInitializer requestInitializer = new HttpCredentialsAdapter(credentials);
 
     return new Compute.Builder(httpTransport, jsonFactory, requestInitializer)
-        .setApplicationName("Google-ComputeSample/0.1")
+        .setApplicationName("VR-Druid-Autoscaler/0.1")
         .build();
   }
 
+  /**
+   * When called, it tries to create envConfig.getTargetWorkers() instances at the time
+   * using the template in envConfig.getinstanceTemplate()
+   */
   @Override
   public AutoScalingData provision()
   {
-    final String project = envConfig.getNodeData().getProjectId();
-    final String zone = envConfig.getNodeData().getZoneName();
-    final int targetWorkers = envConfig.getNodeData().getTargetWorkers();
-    final String instanceGroupManager = envConfig.getNodeData().getInstanceGroupManager();
+    final String project = envConfig.getProjectId();
+    final String zone = envConfig.getZoneName();
+    final int targetWorkers = envConfig.getTargetWorkers();
+    final String instanceTemplate = envConfig.getinstanceTemplate();
 
     try {
+      List<String> instanceIds = new LinkedList<>();
       Compute computeService = createComputeService();
-      Compute.InstanceGroupManagers.Resize request =
-          computeService.instanceGroupManagers().resize(project, zone,
-              instanceGroupManager, targetWorkers);
-
-      Operation response = request.execute();
-
-      Compute.InstanceGroupManagers.ListManagedInstances request2 =
-          computeService
-          .instanceGroupManagers()
-          .listManagedInstances(project, zone, instanceGroupManager);
-
-      InstanceGroupManagersListManagedInstancesResponse response2 = request2.execute();
-      List<ManagedInstance> instances = response2.getManagedInstances();
-      List<String> instanceIds = new ArrayList<>();
-
-      for (ManagedInstance instance : instances) {
-        instanceIds.add(instance.getInstance());
+      for (int i = 0; i < targetWorkers; ++i) {
```

Comment:
I will try to understand if this might be written in a better way instead of a loop